pywinauto/recorder/win32/win32_recorder.py

```diff
@@ -0,0 +1,216 @@
+
+def print_winmsg(msg):
+    pass
+    #print(handleprops.classname(msg.hWnd))
+    print("message:{}".format((msg_id_to_key[msg.message] if msg.message in msg_id_to_key else str(msg.message))))
+
+class Win32Recorder(BaseRecorder):
+    _APPROVED_MESSAGES_LIST = [
+        win32defines.WM_CONTEXTMENU,
+        win32defines.WM_ENTERMENULOOP,
+        win32defines.WM_EXITMENULOOP,
+        # win32defines.WM_GETTITLEBARINFOEX,
+        win32defines.WM_MENUCOMMAND,
+        # win32defines.WM_MENUDRAG,
+        # win32defines.WM_MENUGETOBJECT,
+        # win32defines.WM_MENURBUTTONUP,
+        win32defines.WM_NEXTMENU,
+        # win32defines.WM_UNINITMENUPOPUP,
+        win32defines.WM_COMMAND,
+        292,
+        0x0126,
+        0x0124,
+        0x033F,
+        0x011F,
+        0x0120,
+        win32defines.WM_QUIT,
+        win32defines.WM_NOTIFY,
+        win32defines.WM_KEYUP,
+        win32defines.WM_SETFOCUS,
+        win32defines.WM_COMPAREITEM,
+        # win32defines.WM_DRAWITEM,
+        win32defines.WM_MEASUREITEM,
+    ]
+
+    def __init__(self, app, config, record_props=True, record_focus=False, record_struct=False):
+        super(Win32Recorder, self).__init__(app=app, config=config)
+        if app.backend.name != "win32":
+            raise TypeError("app must be a pywinauto.Application object of 'win32' backend")
+
+        self.last_kbd_hwnd = None
+        self.app = app
+        self.dlg = app[config.cmd]
+        self.listen = False
+        self.hook = win32_hooks.Hook()
+        self.record_props = record_props
+        self.record_focus = record_focus
+        self.record_struct = record_struct
+
+    def _setup(self):
+        try:
+            self.injector = Injector(self.dlg, approved_messages_list = self._APPROVED_MESSAGES_LIST)
+            self.socket = self.injector.socket
+            self.listen = True
+            self.control_tree = ControlTree(self.wrapper, skip_rebuild=True)
+            self._update(rebuild_tree=True, start_message_queue=True)
+        except:
+            self.stop()
+
+    def _cleanup(self):
+        self.listen = False
+        self.hook.stop()
+        self.message_thread.join(1)
+        self.hook_thread.join(1)
+        self.socket.close()
+        self._parse_and_clear_log()
+        self.script += u"app.kill()\n"
+
+    def _pause_hook_thread(self):
+        self.hook.stop()
+        time.sleep(1)
+        self.hook_thread.join(1)
+
+    def _resume_hook_thread(self):
+        self.hook_thread = threading.Thread(target=self.hook_target)
+        self.hook_thread.start()
+
+    def _update(self, rebuild_tree=False, start_message_queue=False):
+        if rebuild_tree:
+            pbar_dlg = ProgressBarDialog(self.control_tree.root.rect if self.control_tree.root else None)
+            pbar_dlg.show()
+
+            self._pause_hook_thread()
+
+            rebuild_tree_thr = threading.Thread(target=self._rebuild_control_tree)
```

Comment:
TODO: Looks like this code is common for both recorders. But it may be not common for Linux/macOS because of `ProgressBarDialog` implementation. We need to think how to make cross-platform implementation of this dialog or make a slot for OS specific implementation (it can be optional or mandatory slot).